File: ftalentcode/L.py
import heapq
import math

class Part:
    def __init__(self, index, deviation):
        self.index = index
        self.deviation = deviation

    def __lt__(self, other):
        return self.deviation < other.deviation

def main():
    n, k = map(float, input().split())
    n = int(n)
    a = list(map(int, input().split()))
    s = list(map(int, input().split()))

    pq = []
    total_deviation = 0.0
    for i in range(n):
        deviation = abs(a[i] - s[i]) / s[i]
        total_deviation += deviation
        heapq.heappush(pq, Part(i, deviation))

    time = 0
    while total_deviation > k:
        part = heapq.heappop(pq)
        i = part.index
        current_deviation = part.deviation

        if a[i] > s[i]:
            a[i] -= 1
        else:
            a[i] += 1

        new_deviation = abs(a[i] - s[i]) / s[i]
        total_deviation = total_deviation - current_deviation + new_deviation
        heapq.heappush(pq, Part(i, new_deviation))
        time += 1

    print(time)

if __name__ == "__main__":
    main()

File: ftalentcode/L.cpp
#include <iostream>
#include <vector>
#include <queue>
#include <cmath>
using namespace std;

const double EPSILON = 1e-9;

struct Part
{
    int index;
    double deviation;
    bool operator<(const Part &other) const
    {
        return deviation > other.deviation;
    }
};

int main()
{
    int n;
    double k;
    cin >> n >> k;
    vector<double> a(n), s(n);
    for (int i = 0; i < n; ++i)
    {
        cin >> a[i];
    }
    for (int i = 0; i < n; ++i)
    {
        cin >> s[i];
    }

    priority_queue<Part> pq;
    double totalDeviation = 0.0;
    for (int i = 0; i < n; ++i)
    {
        double deviation = fabs(a[i] - s[i]) / s[i];
        totalDeviation += deviation;
        pq.push({i, deviation});
    }

    int time = 0;
    while (totalDeviation > k + EPSILON)
    {
        Part part = pq.top();
        pq.pop();
        int i = part.index;
        double currentDeviation = part.deviation;

        if (a[i] > s[i])
        {
            a[i]--;
        }
        else
        {
            a[i]++;
        }

        double newDeviation = fabs(a[i] - s[i]) / s[i];
        totalDeviation = totalDeviation - currentDeviation + newDeviation;
        pq.push({i, newDeviation});
        time++;
    }

    cout << time << endl;
    return 0;
}